pkg/cards/sql.go

```diff
@@ -0,0 +1,117 @@
+package cards
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/readr-media/readr-restful/config"
+	"github.com/readr-media/readr-restful/internal/rrsql"
+)
+
+func (a *newscardAPI) buildGetStmt(args *NewsCardArgs) (query string, values []interface{}) {
+	selectedFields := []string{"newscards.*"}
+	var restricts string
+
+	restricts, restrictVals := args.parse()
+	resultLimit, resultLimitVals := args.parseResultLimit()
+	values = append(values, restrictVals...)
+	values = append(values, resultLimitVals...)
+
+	query = fmt.Sprintf(`
+		SELECT %s FROM newscards %s `,
+		strings.Join(selectedFields, ","),
+		restricts+resultLimit,
+	)
+
+	return query, values
+}
+
+func BuildSyncStmts(postID uint32, cards []NewsCard) (stmts []*rrsql.PipelineStmt, err error) {
+
+	var cardsToBeUpdated, cardsToBeInserted []NewsCard
+	for _, card := range cards {
+		if card.ID == 0 {
+			cardsToBeInserted = append(cardsToBeInserted, card)
+		} else {
+			cardsToBeUpdated = append(cardsToBeUpdated, card)
+		}
+	}
+
+	if postID != 0 {
+		var cardIDs []uint32
+		for _, card := range cardsToBeUpdated {
+			cardIDs = append(cardIDs, card.ID)
+		}
+		stmt, err := buildDeleteStmt(postID, cardIDs)
```

Comment:
I'm not sure why we have to delete cards id if post id is not 0

Reply:
See line 75, the query specifies to delete cards not in the list of cards to be updated.
This means that cards not contained in the parameter `cards` and exists in the DB in the same time should be deleted, according to the spec.